qiskit_ibm_runtime/utils/qctrl.py

```diff
@@ -0,0 +1,143 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qctrl validation functions and helpers."""
+
+import logging
+from typing import Any, Optional, Dict, List
+
+from ..options import Options
+from ..options import EnvironmentOptions, ExecutionOptions, TranspilationOptions, SimulatorOptions
+
+logger = logging.getLogger(__name__)
+
+
+def validate(options: Dict[str, Any]) -> None:
+    """Validates the options for qctrl"""
+
+    # Raise error on bad options.
+    _raise_if_error_in_options(options)
+    # Override options and warn.
+    _warn_and_clean_options(options)
+
+    # Default validation otherwise.
+    TranspilationOptions.validate_transpilation_options(options.get("transpilation"))
+    execution_time = options.get("max_execution_time")
+    if not execution_time is None:
+        if (
+            execution_time < Options._MIN_EXECUTION_TIME
+            or execution_time > Options._MAX_EXECUTION_TIME
+        ):
+            raise ValueError(
+                f"max_execution_time must be between "
+                f"{Options._MIN_EXECUTION_TIME} and {Options._MAX_EXECUTION_TIME} seconds."
+            )
+
+    EnvironmentOptions.validate_environment_options(options.get("environment"))
+    ExecutionOptions.validate_execution_options(options.get("execution"))
+    SimulatorOptions.validate_simulator_options(options.get("simulator"))
+
+
+def _raise_if_error_in_options(options: Dict[str, Any]) -> None:
+    """Checks for settings that produce errors and raise a ValueError"""
+
+    # Fail on resilience_level set to 0
+    resilience_level = options.get("resilience_level", 1)
+    _check_argument(
+        resilience_level > 0,
+        description="Q-CTRL Primitives do not support resilience level 0. Please\
+        set resilience_level to 1 and re-try",
+        arguments={},
+    )
+
+    # Fail on skipping transpilation
+    skip_transpilation = options.get("transpilation", {}).get("skip_transpilation", False)
+    _check_argument(
+        skip_transpilation is False,
+        description="Q-CTRL Primitives skipping transpilation. Please\
+        set skip_transpilation to False and re-try",
+        arguments={},
+    )
+
+
+def _warn_and_clean_options(options: Dict[str, Any]) -> None:
+    """
+    Validate and update transpilation settings
+    """
+    # Issue a warning and override if any of these setting is not None
+    # or a different value than the default below
+    expected_options = {
+        "optimization_level": 3,
+        "transpilation": {"approximation_degree": 0},
+        "resilience_level": 1,
+        "resilience": {
+            "noise_amplifier": None,
+            "noise_factors": None,
+            "extrapolator": None,
+        },
```

Comment:
Related to #1048 and #1049, I'd prefer if we set the proper defaults when the Q-Ctrl strategy is selected instead of overriding them like this. I haven't been able to test the full job flow yet but the rest of PR looks good to me

Reply:
We'll need this to be able to update settings on improper user input, even after meaningful defaults.